.github/workflows/run-tests.yml

```diff
@@ -16,48 +16,57 @@ jobs:
   lint:
     name: Lint
     runs-on: ubuntu-latest
+    env:
+      VOLTA_FEATURE_PNPM: 1
 
     steps:
       - uses: actions/checkout@v4
       - uses: volta-cli/action@v4
+      - id: pnpm-cache-path
+        run: echo "pnpm_cache_path=$(pnpm store path --silent)" >> $GITHUB_OUTPUT
       - uses: actions/cache@v4
-        id: node-cache
         with:
-          path: "**/node_modules"
-          key: ${{ runner.os }}-ci-yarn-${{ hashFiles('**/yarn.lock') }}
+          path: ${{ steps.pnpm-cache-path.outputs.pnpm_cache_path }}
+          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
       - name: Install Dependencies
-        if: steps.node-cache.outputs.cache-hit != 'true'
-        run: yarn --frozen-lockfile
+        run: pnpm install
```

Comment:
Verified that the cache is working correctly by reviewing a recent run:

![image](https://github.com/user-attachments/assets/4cc1bd55-d504-4f77-b173-868b3be23fd5)